db/core/htrie.c

```diff
@@ -365,8 +438,8 @@ tdb_alloc_blk(TdbHdr *dbh)
 	if (unlikely(TDB_HTRIE_OFF(dbh, e) == dbh->dbsz)) {
 		/* We do this set because we skip the last page in extent. */
 		atomic64_set(&dbh->nwb, dbh->dbsz);
-		TDB_ERR("out of free space\n");
-		return 0;
+		dbh->oom = true;
```

Comment:
It seems that we never set dbh->oom to false.  So ig it was set one time we change our allocation strategy for all time. Is it ok?